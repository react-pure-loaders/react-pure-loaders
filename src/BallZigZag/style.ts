import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';

const amount = '30px';

const ballZig = keyframes({
  '100%': {
    transform: 'translate(0, 0)',
  },
  '33%': {
    transform: `translate(-${parseInt(amount, 10) / 2}px, -${amount})`,
  },
  '66%': {
    transform: `translate(${parseInt(amount, 10) / 2}px, -${amount})`,
  },
});

const ballZag = keyframes({
  '100%': {
    transform: 'translate(0, 0)',
  },
  '33%': {
    transform: `translate(${parseInt(amount, 10) / 2}px, ${amount})`,
  },
  '66%': {
    transform: `translate(-${parseInt(amount, 10) / 2}px, ${amount})`,
  },
});

export default (color?: string) => ({
  position: 'relative' as 'relative',
  transform: `translate(-${parseInt(amount, 10) / 2}px, -${parseInt(amount, 10) / 2}px)`,
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    left: '-7px',
    marginLeft: (parseInt(amount, 10) / 2),
    position: 'absolute' as 'absolute',
    top: '4px',

    '&:first-of-type': {
      animation: `${ballZig} 0.7s 0s infinite linear`,
    },

    '&:last-child': {
      animation: `${ballZag} 0.7s 0s infinite linear`,
    },
  },
});
