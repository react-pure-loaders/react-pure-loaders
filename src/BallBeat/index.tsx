/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
