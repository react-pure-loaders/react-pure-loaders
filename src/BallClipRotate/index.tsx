/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
