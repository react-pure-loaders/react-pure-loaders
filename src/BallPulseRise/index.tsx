/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
