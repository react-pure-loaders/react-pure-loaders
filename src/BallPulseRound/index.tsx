/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
