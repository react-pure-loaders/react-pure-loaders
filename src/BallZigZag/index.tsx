/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
