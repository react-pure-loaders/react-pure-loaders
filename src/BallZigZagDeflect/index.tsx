/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
