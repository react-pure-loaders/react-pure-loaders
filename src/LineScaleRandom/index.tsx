/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
