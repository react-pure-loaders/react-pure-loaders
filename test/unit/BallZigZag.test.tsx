import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallZigZag from '../../src/BallZigZag';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallZigZag>', () => {
    afterEach(cleanup);

    test('BallZigZag should match snapshot', () => {
        const { container } = render(<BallZigZag loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallZigZag should have default color', () => {
        const { container } = render(<BallZigZag loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('BallZigZag should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallZigZag color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallZigZag should have no children', () => {
        const { container } = render(<BallZigZag loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallZigZag should have three children', () => {
        const { container } = render(<BallZigZag loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(3);
    });
});
