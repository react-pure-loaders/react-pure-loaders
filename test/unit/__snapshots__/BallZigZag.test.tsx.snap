// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallZigZag> BallZigZag should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: translate(0, 0);
    -moz-transform: translate(0, 0);
    -ms-transform: translate(0, 0);
    transform: translate(0, 0);
  }

  33% {
    -webkit-transform: translate(-15px, -30px);
    -moz-transform: translate(-15px, -30px);
    -ms-transform: translate(-15px, -30px);
    transform: translate(-15px, -30px);
  }

  66% {
    -webkit-transform: translate(15px, -30px);
    -moz-transform: translate(15px, -30px);
    -ms-transform: translate(15px, -30px);
    transform: translate(15px, -30px);
  }
}

@keyframes animation-1 {
  100% {
    -webkit-transform: translate(0, 0);
    -moz-transform: translate(0, 0);
    -ms-transform: translate(0, 0);
    transform: translate(0, 0);
  }

  33% {
    -webkit-transform: translate(15px, 30px);
    -moz-transform: translate(15px, 30px);
    -ms-transform: translate(15px, 30px);
    transform: translate(15px, 30px);
  }

  66% {
    -webkit-transform: translate(-15px, 30px);
    -moz-transform: translate(-15px, 30px);
    -ms-transform: translate(-15px, 30px);
    transform: translate(-15px, 30px);
  }
}

.emotion-0 {
  position: relative;
  -webkit-transform: translate(-15px, -15px);
  -moz-transform: translate(-15px, -15px);
  -ms-transform: translate(-15px, -15px);
  transform: translate(-15px, -15px);
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  left: -7px;
  margin-left: 15px;
  position: absolute;
  top: 4px;
}

.emotion-0>div:first-of-type {
  -webkit-animation: animation-0 0.7s 0s infinite linear;
  animation: animation-0 0.7s 0s infinite linear;
}

.emotion-0>div:last-child {
  -webkit-animation: animation-1 0.7s 0s infinite linear;
  animation: animation-1 0.7s 0s infinite linear;
}

<div
  class="emotion-0"
>
  <div />
  <div />
</div>
`;
